internal/evaluation_target/application_snapshot_image/application_snapshot_image.go

```diff
@@ -247,14 +247,39 @@ func (a *ApplicationSnapshotImage) ValidateAttestationSignature(ctx context.Cont
 
 	// Extract the signatures from the attestations here in order to also validate that
 	// the signatures do exist in the expected format.
-
-	for _, att := range layers {
-		sp, err := attestation.SLSAProvenanceFromSignature(att)
+	for _, sig := range layers {
+		att, err := attestation.UntypedProvenanceFromSignature(sig)
 		if err != nil {
-			log.Debugf("Ignoring non SLSA Provenance attestation: %s", err)
-			continue
+			return fmt.Errorf("Unable to parse untyped provenance: %w", err)
+		}
+		t := att.PredicateType()
+		switch t {
+		case attestation.PredicateSLSAProvenance:
+			// SLSAProvenanceFromSignature does the payload extraction
+			// and decoding that was done in UntypedProvenanceFromSignature
+			// over again. We could refactor so we're not doing that twice,
+			// but it's not super important IMO.
+			sp, err := attestation.SLSAProvenanceFromSignature(sig)
+			if err != nil {
+				return fmt.Errorf("Unable to parse as SLSA v0.2: %w", err)
+			}
+
+			// Keep for further use
+			a.attestations = append(a.attestations, sp)
+
+		case attestation.PredicateSpdxDocument:
+			// Todo: Try unmarshaling to confirm the format is valid
+
+			// Keep for further use
+			a.attestations = append(a.attestations, att)
+
+		default:
+			// Todo: We should probably keep rather than discard attestations with
+			// an unknown predicateType
+			log.Debugf("Ignoring attestation with predicateType: %s", t)
```

Comment:
Would make sense for future proofing to not discard here

Reply:
Yeah, we could choose to be bold and do it now.

Reply:
Done.